### PR TITLE
remove src from windoors

### PR DESCRIPTION
**Pull request by SonixApache**
CTRL+H `src.` into nothing
compiled and tested, opened and closed a windoor, dissassembled and re-assembled, hit it a bit. All seems in order.

windoor code still looks and feels bad but at least i won't cringe every time i write something and see a `src.x` two lines over

definitely going to conflict with smartglass

### CODE REVIEW

**Comment by N3X15** [comment]
Smartglass came first, so that'll be merged first.

**Comment by JustSumBody** [comment]
nonstandard formatting change

**Comment by Exxion** [comment]
It is but honestly who cares
Removing it from the entire codebase haphazardly was a stupid idea, because it DOES sometimes break things if you remove it
But for a single file, go ahead, I guess, as long as you test it

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -25,9 +25,9 @@
 
 /obj/machinery/door/window/New()
 	..()
-	if ((istype(src.req_access) && src.req_access.len) || istext(req_access))
-		src.icon_state = "[src.icon_state]"
-		src.base_state = src.icon_state
+	if ((istype(req_access) && req_access.len) || istext(req_access))
+		icon_state = "[icon_state]"
```
Comment (Intigracy) [inline code/game/machinery/doors/windowdoor.dm:31]: lmao what even is this doing

**Comment by SonixApache** [comment]
code is seven year old from Erthilo/svn stuff

**Comment by Shadowmech88** [comment]
>because it DOES sometimes break things if you remove it

Are there some functions in BYOND that *require* `.src` to be performed?

**Comment by unid15** [comment]
If you have a local variable and an object-level variable, src is used to differentiate between them, e.g. if you're defining a New() proc and you create a local variable var/name = "Alpha", you use [src.name] to get the object's name, and [name] to get "Alpha"

**Comment by N3X15** [comment]
And now has actual conflicts.

**Comment by Shadowmech88** [comment]
Is there any reason to intentionally give a local variable the same name as an object-level variable?

**Comment by Exxion** [comment]
One marginal performance increase, but other than that, no
But that doesn't mean nobody did it
(They did)

**Pull request merged**